Automattic/jetpack
# Blocks: Show the Launchpad on save modal inside the Post Editor for all flows

Fixes [LaunchPad - adapt and enable the back-to-launchpad modal in the post/page editor](https://github.com/Automattic/wp-calypso/issues/72093) and [LaunchPad - editor 'back-to-launchpad' modal tracking events](https://github.com/Automattic/wp-calypso/issues/72097)

## Proposed changes:
This PR provides a generic modal for all Launchpad flows to allow the user to go back to the Launchpad screen from the Post Editor once they have saved their progress. It also adds tracking events

<img width="874" alt="image" src="https://user-images.githubusercontent.com/20927667/214405253-d843c1c0-371d-449e-87e0-0a1dda1d4b82.png">

### Other information:


## Jetpack product discussion
paYE8P-1Kg-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

### Testing on WPCOM
1. Sandbox `public-api.wordpress.com` along with a Launchpad enabled site
2. Run `bin/jetpack-downloader test jetpack add/launchpad_save_modal_post_editor` on your sandbox
3. Go to the Post Editor to write a New post
4. Start making changes but don't click publish yet and wait for the auto-save functionality to kick in. Verify the modal doesn't show up on auto-save. (This will be true only when the post is not published yet, it will show up on auto-save when you making changes to a published post)
5. Click publish. You should see the modal appear.
6. Keep making changes and updating the post. You should see the modal appear every time you update the post.
5. Test with different flows - `newsletter`, `link-in-bio`, `free`, `videopress`.
7. Use [Tracks Vig ](https://github.com/Automattic/tracks-chrome-extension) to test the tracks events:
• Clicking the close button -> `jetpack_launchpad_save_modal_close`
• Clicking the Back to Edit -> `jetpack_launchpad_save_modal_back_to_edit`
• Clicking the Next Button -> `jetpack_launchpad_save_modal_next_steps`
8. Do a regression test on the Site Editor modal following the testing instructions in the PR: [Blocks: Show the Launchpad on save modal inside the Site Editor for all flows](https://github.com/Automattic/jetpack/pull/28503)
